agent/src/heapstats-engines/snapShotContainer.cpp

```diff
@@ -502,9 +506,8 @@ void TSnapShotContainer::mergeChildren(void) {
             }
 
             prevCounter = counter;
+            counter = counter->next;
           }
-
-          counter = counter->next;
```

Comment:
This statement will be called when `counter` is free'ed in above.
We should avoid this situation.